mysql-test/main/backup_locks.result

```diff
@@ -39,6 +39,27 @@ MDL_INTENTION_EXCLUSIVE	Schema metadata lock	test
 select * from t1;
 ERROR 40001: Deadlock found when trying to get lock; try restarting transaction
 backup unlock;
+#
+# MDEV-20945 BACKUP UNLOCK assertion failures.
+#
+# Scenario 1.
+CREATE TABLE t1(a INT);
+BACKUP LOCK t1;
+FLUSH TABLE t1 WITH READ LOCK;
+UNLOCK TABLES;
+BACKUP UNLOCK;
+DROP TABLE t1;
+# Scenario 2.
+CREATE TABLE t1(a INT);
+CREATE TABLE t2(b INT);
+LOCK TABLES t2 AS a2 WRITE;
+BACKUP LOCK t1;
+UNLOCK TABLES;
+INSERT INTO t1 VALUES(0);
+BACKUP UNLOCK;
+DROP TABLE t1;
+DROP TABLE t2;
```

Comment:
In jira there are more other examples of usage on non-existing tables/databases, would be good to cover them also.

Reply:
Thanks for the suggestion. Added all snippets I found in Jira's MDEV-20495 as test cases.
Also while checking scenarios I realized that some of the crashes were happening during shutdown, so added server restarts to the test too.